src/components/BranchMenu.tsx

```diff
@@ -0,0 +1,315 @@
+import * as React from 'react';
+import { classes } from 'typestyle';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ClearIcon from '@material-ui/icons/Clear';
```

Comment:
Is it ok to import icons from third-party library or is it better to ship all icons except those shared with JupyterLab? @telamonian what do you think?

Reply:
We could eventually move to using the [`jp-CloseIcon`](https://github.com/jupyterlab/jupyterlab/blob/3cb64ca6885358e7f9623f518518bb3f9de017bc/packages/ui-components/style/icons.css#L318) once it is ready for use. For now, I don't think importing an icon should be a blocker. 

In line with [elsewhere](https://github.com/jupyterlab/jupyterlab-git/pull/518#discussion_r370909318), I could, however, just get the SVG and leverage CSS. Here, I opted for convenience.

Reply:
Hmm, good question. For the purposes of this PR, and probably in general, I think extension devs using a handful of icons from a third party package is fine (I know that some people are *very* attached to font-awesome, for example).

I do think you're right that the best practice would be to ship the icons ourselves. There's a boatload of icon functionality (theming, replacement, reuse, etc) that in general just won't work with 3rd party icons. I'm actually hard at work right now putting the finishing touches on `LabIcon` (formerly `JLIcon`), which will be the new icon system for jlab 2.0. It should be finished in the next day or two (or rather, it better well be, or else I'll end up blocking the 2.0rc release :sweat_smile:). Once that's done with I'll start working on adding to the [jupyterlab-git-for-jlab-2.0](https://github.com/jupyterlab/jupyterlab-git/pull/520) PR that @blink1073 started, with the goal of moving all of our icons over to `LabIcon` (including finding/making equivalents of the material-ui icons).

Reply:
Awesome! Thanks for the update @telamonian!